src/main/java/com/volmit/iris/core/TreeManager.java

```diff
@@ -0,0 +1,217 @@
+     * <br>5. Exit if none are found, cancel event if one or more are.</br>
+     * @param event Checks the given event for sapling overrides
+     */
+    @EventHandler
+    public void onStructureGrowEvent(StructureGrowEvent event) {
+
+        Iris.debug(this.getClass().getName() + " received a structure grow event");
+
+        // Must be iris world
+        if (!IrisWorlds.isIrisWorld(event.getWorld())) {
+            Iris.debug(this.getClass().getName() + " passed it off to vanilla since not an Iris world");
+            return;
+        }
+
+        // Get world access
+        IrisAccess worldAccess;
+        try {
+            worldAccess = Objects.requireNonNull(IrisWorlds.access(event.getWorld()));
+        } catch (Throwable e) {
+            Iris.debug(this.getClass().getName() + " passed it off to vanilla because could not get IrisAccess for this world");
+            Iris.reportError(e);
+            return;
+        }
+
+        // Return null if not enabled
+        if (!worldAccess.getCompound().getRootDimension().getTreeSettings().isEnabled()) {
+            Iris.debug(this.getClass().getName() + " cancelled because tree overrides are disabled");
+            return;
+        }
+
+        Iris.debug("Sapling grew @ " + event.getLocation() + " for " + event.getSpecies().name() + " usedBoneMeal is " + event.isFromBonemeal());
+
+        // Calculate size, type & placement
+        IrisTreeType type = IrisTreeType.fromTreeType(event.getSpecies());
+        KMap<IrisTreeSize, KList<KList<Location>>> sizes = IrisTreeSize.getValidSizes(event.getLocation());
+        KList<IrisTreeSize> keys = sizes.k();
+
+        // Check if any were returned
+        if (keys.isEmpty()) {
+            Iris.debug(this.getClass().getName() + " found no matching sapling sizes for the grow event, which should be impossible (considering ONE is an option)");
+            return;
+        }
+
+        // Find best object placement based on sizes
+        IrisObjectPlacement placement = null;
+        IrisTreeSize bestSize = null;
+        while (placement == null && keys.isNotEmpty()){
+            bestSize = IrisTreeSize.bestSizeInSizes(keys);
+            keys.remove(bestSize);
+            placement = getObjectPlacement(worldAccess, event.getLocation(), type, bestSize);
+        }
+
+        // If none were found, just exit
+        if (placement == null) {
+            Iris.debug(this.getClass().getName() + " had options but did not manage to find objectPlacements for them");
+            return;
+        }
+
+        // Cancel the placement
+        event.setCancelled(true);
+
+        // Delete existing saplings
+        sizes.get(bestSize).forEach(row -> row.forEach(location -> location.getBlock().setType(Material.AIR)));
+
+        // Get object from placer
+        IrisObject f = worldAccess.getData().getObjectLoader().load(placement.getPlace().getRandom(RNG.r));
+
```

Comment:
You need to actually tell bukkit what you are doing here so another plugin can find all your changed blocks. To do this make a statelist

```java
event.getBlockStates().clear();
```